iothub/device/src/ModernDotNet/HsmAuthentication/Transport/HttpUdsMessageHandler.cs

```diff
@@ -42,9 +42,14 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
 
         private async Task<Socket> GetConnectedSocketAsync()
         {
-            var endpoint = new UnixDomainSocketEndPoint(_providerUri.LocalPath);
             Socket socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+#if NET451 || NET472 || NETSTANDARD2_0
+            var endpoint = new UnixDomainSocketEndPoint(_providerUri.LocalPath);
             await socket.ConnectAsync(endpoint).ConfigureAwait(false);
+#else
+            var endpoint = new System.Net.Sockets.UnixDomainSocketEndPoint(_providerUri.LocalPath);
```

Comment:
Isn't `System.Net.Sockets` already in the usings list at the top?

Reply:
What makes this type different than the one on 48?

Reply:
To highlight the difference between using our own impl for <ns2.1 and using System.Net impl for others, we should probably either use a different alias for each UnixDomainSocketEndPoint implementation import or use the full namespace when referencing both of the types.

Reply:
I'd agree. In the very least, use the fully-qualified namespace for our own impl too, or alias ours.

Reply:
And a code comment more or less what is in this PR description that explains why we have this platform-based code fork.